gtsam/sfm/SfmData.h

```diff
@@ -0,0 +1,195 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file SfmData.h
+ * @date January 2022
+ * @author Frank dellaert
+ * @brief Data structure for dealing with Structure from Motion data
+ */
+
+#pragma once
```

Comment:
Good idea to split this up. The `slam/dataset.cpp` file was getting massive and was hard to follow.